### PR TITLE
Upgrade React to v17.

### PR DESCRIPTION
**Pull request by peterp**
This upgrades RWJS to React v17. E2E tests are running fine.

### CODE REVIEW

**Comment by peterp** [comment]
Stupid tests. Why you gotta ruin my fun time!

**Comment by dac09** [comment]
Could we please hold this till 0.27 (may affect prerender)

**Comment by thedavidprice** [comment]
HODL

**Review by thedavidprice — CHANGES_REQUESTED** [review]
E2E = ✅
`rw storybook` = ✅
`rw test` = ⛔️

I was able to run the test command using the PR Packages installation. Tests are running correctly but throwing an error that might be due to React upgrade change. Here's the [React v17 release notes](https://reactjs.org/blog/2020/08/10/react-v17-rc.html).

And here's one of two similar errors I'm hitting with tests (I created the project initially using the E2E script and can confirm the resulting tests should all be passing).

```bash
 FAIL   web  web/src/pages/BlogPostPage/BlogPostPage.test.js
  - Console

    console.error
      Error: Uncaught [Error: You must register a useQuery hook via the `GraphQLHooksProvider`]
...

    console.error
      The above error occurred in the <Query> component:
          in Query
          in Unknown (created by BlogPostPage)
          in main (created by BlogLayout)
          in BlogLayout (created by BlogPostPage)
          in BlogPostPage
          in LocationProvider (created by MockProviders)
          in FlashProvider (created by RedwoodApolloProvider)
          in GraphQLHooksProvider (created by RedwoodApolloProvider)
          in ApolloProvider (created by ApolloProviderWithFetchConfig)
          in ApolloProviderWithFetchConfig (created by RedwoodApolloProvider)
          in FetchConfigProvider (created by RedwoodApolloProvider)
          in RedwoodApolloProvider (created by MockProviders)
          in AuthProvider (created by MockProviders)
          in MockProviders (created by wrapper)
          in wrapper
...
- BlogPostPage › renders successfully

    expect(received).not.toThrow()

    Error name:    "Error"
    Error message: "You must register a useQuery hook via the `GraphQLHooksProvider`"

           6 |   it('renders successfully', () => {
           7 |     expect(() => {
        >  8 |       render(<BlogPostPage />)
             |       ^
           9 |     }).not.toThrow()
          10 |   })
          11 | })

          at Object.useQuery (node_modules/@redwoodjs/web/dist/components/GraphQLHooksProvider.js:19:11)
          at useQuery (node_modules/@redwoodjs/web/dist/components/GraphQLHooksProvider.js:52:57)
          at Query (node_modules/@redwoodjs/web/dist/components/withCellHOC.js:28:53)
...
          at Object.toThrow (node_modules/expect/build/index.js:338:21)
          at Object.<anonymous> (web/src/pages/BlogPostPage/BlogPostPage.test.js:9:12)

       7 |     expect(() => {
       8 |       render(<BlogPostPage />)
    >  9 |     }).not.toThrow()
         |            ^
      10 |   })
      11 | })
      12 |

      at Object.<anonymous> (src/pages/BlogPostPage/BlogPostPage.test.js:9:12)
```

The other is for Home Page:
```bash
...
  ● HomePage › renders successfully

    expect(received).not.toThrow()

    Error name:    "Error"
    Error message: "You must register a useQuery hook via the `GraphQLHooksProvider`"

           6 |   it('renders successfully', () => {
           7 |     expect(() => {
        >  8 |       render(<HomePage />)
             |       ^
           9 |     }).not.toThrow()
          10 |   })
          11 | })
...
```

**Comment by peterp** [comment]
@thedavidprice I figured out the problem with the tests. If you run `yarn rwt cpw` and copy the packages they work again. I think the problem is that jest can't figure out where it's running from...

File: packages/auth/src/authClients/nhost.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ export const nhost = (client: Nhost): AuthClient => {
       return await client.auth.register({
         email,
         password,
-        registrationOptions: { userData: { display_name: email } },
+        options: { userData: { display_name: email } },
```
Comment (peterp) [inline packages/auth/src/authClients/nhost.ts:80]: @Tobbe @nunopato I think I may have upgraded a package or something that changed this. I checked NHosts library implementation and it seems to match.

https://github.com/nhost/nhost-js-sdk/blob/master/src/Auth.ts#L104

**Pull request merged**